From: Yajun Deng <yajun.deng@linux.dev>
Subject: mm: simplify page_zone() and get_deferred_split_queue() with page_pgdat()
Date: Tue, 1 Nov 2022 10:32:19 +0800

page_zone() is too long, we can shorten it by page_pgdat().

There is no need define pgdat variable if memcg isn't NULL, so use
page_pgdat() to simplify get_deferred_split_queue().

Link: https://lkml.kernel.org/r/20221101023219.752365-1-yajun.deng@linux.dev
Signed-off-by: Yajun Deng <yajun.deng@linux.dev>
Cc: Mike Kravetz <mike.kravetz@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mm.h |    8 ++++----
 mm/huge_memory.c   |    7 ++-----
 2 files changed, 6 insertions(+), 9 deletions(-)

--- a/include/linux/mm.h~mm-simplify-page_zone-and-get_deferred_split_queue-with-page_pgdat
+++ a/include/linux/mm.h
@@ -1448,14 +1448,14 @@ static inline void page_kasan_tag_reset(
 
 #endif /* CONFIG_KASAN_SW_TAGS || CONFIG_KASAN_HW_TAGS */
 
-static inline struct zone *page_zone(const struct page *page)
+static inline pg_data_t *page_pgdat(const struct page *page)
 {
-	return &NODE_DATA(page_to_nid(page))->node_zones[page_zonenum(page)];
+	return NODE_DATA(page_to_nid(page));
 }
 
-static inline pg_data_t *page_pgdat(const struct page *page)
+static inline struct zone *page_zone(const struct page *page)
 {
-	return NODE_DATA(page_to_nid(page));
+	return &page_pgdat(page)->node_zones[page_zonenum(page)];
 }
 
 static inline struct zone *folio_zone(const struct folio *folio)
--- a/mm/huge_memory.c~mm-simplify-page_zone-and-get_deferred_split_queue-with-page_pgdat
+++ a/mm/huge_memory.c
@@ -562,19 +562,16 @@ pmd_t maybe_pmd_mkwrite(pmd_t pmd, struc
 static inline struct deferred_split *get_deferred_split_queue(struct page *page)
 {
 	struct mem_cgroup *memcg = page_memcg(compound_head(page));
-	struct pglist_data *pgdat = NODE_DATA(page_to_nid(page));
 
 	if (memcg)
 		return &memcg->deferred_split_queue;
 	else
-		return &pgdat->deferred_split_queue;
+		return &page_pgdat(page)->deferred_split_queue;
 }
 #else
 static inline struct deferred_split *get_deferred_split_queue(struct page *page)
 {
-	struct pglist_data *pgdat = NODE_DATA(page_to_nid(page));
-
-	return &pgdat->deferred_split_queue;
+	return &page_pgdat(page)->deferred_split_queue;
 }
 #endif
 
_
